File: app/src/main/java/lt/markmerkk/utils/FirstSettings.java
package lt.markmerkk.utils;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.util.Properties;
import lt.markmerkk.Main;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Created by mariusmerkevicius on 11/24/15.
 * First time launch settings
 */
public class FirstSettings extends BaseSettings {
  Logger logger = LoggerFactory.getLogger(FirstSettings.class);
  public static final String PROPERTIES_FILE = "first.properties";

  boolean firstLaunch = true;

  //region Core

  public boolean isFirst() {
    return firstLaunch;
  }

  @Override
  String propertyPath() {
    return PROPERTIES_FILE;
  }

  @Override
  void onLoad(Properties properties) {
    String firstProperty = properties.getProperty("first", "true");
    firstLaunch = !"false".equals(firstProperty);
  }

  @Override
  void onSave(Properties properties) {
    properties.put("first", "false");
  }

  //endregion

}

File: app/src/main/java/lt/markmerkk/utils/FirstSettings.kt
package lt.markmerkk.utils

import org.slf4j.LoggerFactory
import java.util.*

/**
 * Created by mariusmerkevicius on 11/24/15.
 * First time launch settings
 */
class FirstSettings : BaseSettings() {
    var isFirst = true
        private set

    override fun propertyPath(): String {
        return PROPERTIES_FILE
    }

    override fun onLoad(properties: Properties) {
        val firstProperty = properties.getProperty("first", "true")
        isFirst = "false" != firstProperty
    }

    override fun onSave(properties: Properties) {
        properties.put("first", "false")
    }

    companion object {
        var logger = LoggerFactory.getLogger(FirstSettings::class.java)
        val PROPERTIES_FILE = "first.properties"
    }

}
